lib/surface/components/button.ex

```diff
@@ -0,0 +1,183 @@
+defmodule Surface.Components.Button do
+  @moduledoc """
+  Generates a button that uses a regular HTML form to submit to the given URL.
+
+  Useful to ensure that links that change data are not triggered by search engines and other spidering software.
+
+  Provides similar capabilities to Phoenix's built-in `button/2` function.
+
+  Options `label` and `class` can be set directly and will override anything in `opts`.
+  All other options are forwarded to the underlying <button> tag.
+
+  ## Examples
+  ```
+  <Button
+    label="user"
+    to="/users/1"
+    class="is-danger"
+    opts={{ method: :delete, data: [confirm: "Really?"] }}
+  />
+
+  <Button
+    to="/users/1"
+    class="is-link"
+  >
+    <span>user</span>
+  </Button>
+  ```
+  """
+
+  use Surface.Component
+
+  @doc "The page to link to"
+  prop to, :any, required: true
+
+  @doc "The method to use with the button"
+  prop method, :atom, default: :post
+
+  @doc "Id to apply to the button"
+  prop id, :string
+
+  @doc "Class or classes to apply to the button"
+  prop class, :css_class
+
+  @doc """
+  The label for the generated `<button>` element, if no content (default slot) is provided.
+  """
+  prop label, :string
+
+  @doc "Triggered on click"
+  prop click, :event
```

Comment:
Shouldn't we add the others usual events like the ones we find in the `Surface.Components.Form.Submit` component?
```elixir
      @doc "Triggered when the component loses focus"
      prop blur, :event

      @doc "Triggered when the component receives focus"
      prop focus, :event

      @doc "Triggered when the component receives click"
      prop capture_click, :event

      @doc "Triggered when a button on the keyboard is pressed"
      prop keydown, :event

      @doc "Triggered when a button on the keyboard is released"
      prop keyup, :event
```
Moreover, WDYT of renaming `click` to `capture_click` to be consistent in the whole codebase? In that case, we will also modify the `click` event in the `Link` component.

Reply:
Yes makes sense, somehow missed these changes.

It seems that in `Surface.Components.Form.Submit` these events are defined, but never actualy used. In `Surface.Components.Form.Reset` they're are defined and used. Is there a reason for this or just an oversight?

Reply:
> It seems that in `Surface.Components.Form.Submit` these events are defined, but never actualy used. In `Surface.Components.Form.Reset` they're are defined and used. Is there a reason for this or just an oversight?

Probably an oversight.